packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -27,9 +30,123 @@ def _ensure_routable():
         r = requests.post(proxy_uri, data=cmd)
         logging.info('Requests Response: %s', repr(r.json()))
         assert r.json()['status'] == 0
+        return json.loads(r.json()['output'])
     return _ensure_routable
 
 
+def vip_test(dcos_api_session, vip, vipaddr, samehost, vipnet, proxynet):
+    teststr = 'vip={} vipaddr={} samehost={} vipnet={} proxynet={}'.format(vip, vipaddr, samehost, vipnet,
+                                                                           proxynet)
+    logging.info('VIP_TEST START {}'.format(teststr))
+    agents = list(dcos_api_session.slaves)
+    # make sure we can reproduce
+    random.seed(vip)
+    random.shuffle(agents)
+    host1 = agents[0]
+    host2 = agents[0]
+    if not samehost:
+        host2 = agents[1]
```

Comment:
@mellenburg can you explain why it will fail?  it should run with more than two private agents, and distribute across all the private agents randomly.